### PR TITLE
rgw admin: disable stale instance delete in a multiste env

### PR DESCRIPTION
**Pull request by theanalyst**
Disables stale instance deletion in a multisite env

Fixes: https://tracker.ceph.com/issues/38658

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by theanalyst** [comment]
Unsure whether we need to allow a force flag, since listing works I assume someone doing deletion will really know what they are doing, enough to find the instances from the list and delete them manually.

**Comment by cbodley** [comment]
how about requiring the force flag for `stale-instances list`, with an error message saying that the stale instances it identifies are unlikely to be from resharding?

**Comment by theanalyst** [comment]
updated, also created a tracker issue for backports

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -7836,13 +7836,25 @@ int main(int argc, const char **argv)
...
+
    ret = RGWBucketAdminOp::list_stale_instances(store, bucket_op,f);
    if (ret < 0) {
      cerr << "ERROR: listing stale instances" << cpp_strerror(-ret) << std::endl;
    }
  }
 
  if (opt_cmd == OPT_RESHARD_STALE_INSTANCES_DELETE) {
+   if (!store->svc.zone->can_reshard()) {
+     cerr << "Resharding disabled in a multisite env Stale instances are not safe to be deleted." << std::endl;
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc]: period between `env. Stale`?

**Comment by theanalyst** [comment]
rerun: http://pulpito.ceph.com/abhi-2019-03-15_08:00:15-rgw-wip-abhi-testing-2019-03-14-2228-distro-basic-smithi/ looks clean

**Comment by theanalyst** [comment]
Can this go in as it is, since the docs already merged with the other PR. I guess another pr to revamp/move docs under troubleshooting makes sense.

**Pull request merged**